AI/Nullkiller/AIGateway.cpp

```diff
@@ -592,7 +592,7 @@ void AIGateway::showBlockingDialog(const std::string & text, const std::vector<C
 
 					logAi->trace("Guarded object query hook: %s by %s danger ratio %f", target.toString(), hero.name, ratio);
 
-					if(text.find("guarded") >= 0 && (dangerUnknown || dangerTooHigh))
+					if(text.find("guarded") && (dangerUnknown || dangerTooHigh))
```

Comment:
This is wrong. Find method returns npos for non-success

Reply:
But npos is a positive number. So both old and new code are wrong

Reply:
yep, correct way is `!= std::string::npos`